### ISSUE TITLE
Feature request: Generate WSDL as interface

### ISSUE THREAD

**Issue by Eyal-Shalev** [wsdl2phpgenerator/wsdl2phpgenerator#256]
## Motivation
Adding another abstruction layer between the WSDL class implementation and the WSDL specification.

## Solution
Add a configurable option to the Generator that will create interfaces instead of classes

**Comment by garyee** [comment]
What is the goal of the feature?

**Comment by Eyal-Shalev** [comment]
To allow the implementation to be independent of the wsdl which may need to
be changed and rebuilt.

On Tue, Jan 19, 2016, 17:51 Gerald Meier <notifications@github.com> wrote:

> What is the goal of the feature?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/issues/256#issuecomment-172895078>
> .
>
-- 

Regards,
Eyal Shalev

**Comment by garyee** [comment]
+1

**Comment by kasperg** [comment]
I agree that would be interesting. As I see it the only classes that need interfaces is [the generated Service class] (https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/master/src/Service.php#L137). All other classes should be DTOs without external dependencies and should not require interfaces.

Our current custom code generation library does not support interfaces. If anyone wants to take a stab at this I recommend piggybacking on the migration to Zend Code (#255).
